lib/matplotlib/tests/test_rcparams.py

```diff
@@ -422,3 +422,64 @@ def test_rcparams_reset_after_fail():
                 pass
 
         assert mpl.rcParams['text.usetex'] is False
+
+
+def test_if_rctemplate_is_up_to_date():
+    # This tests if the matplotlibrc.template file
+    # contains all valid rcParams.
+    dep1 = mpl._all_deprecated
+    dep2 = mpl._deprecated_set
+    deprecated = list(dep1.union(dep2))
+    #print(deprecated)
+    path_to_rc = mpl.matplotlib_fname()
+    with open(path_to_rc, "r") as f:
+        rclines = f.readlines()
+    missing = {}
+    for k,v in mpl.defaultParams.items():
+        if k[0] == "_":
```

Comment:
Maybe join:
~~~
if k.startswith(' ') or k in deprecated or 'verbose' in k:
    continue
~~~